test/e2e/common_suite_test.go

```diff
@@ -254,11 +281,71 @@ func newTestCase(t *testing.T, testName string, assert CloudAssert, assessMessag
 		assert:        assert,
 		assessMessage: assessMessage,
 		podState:      v1.PodRunning,
+		podWatcher:    false,
 	}
 
 	return testCase
 }
 
+func reverseSlice(slice []string) []string {
+	length := len(slice)
+	for i := 0; i < length/2; i++ {
+		slice[i], slice[length-i-1] = slice[length-i-1], slice[i]
+	}
+	return slice
+}
+func watchImagePullTime(ctx context.Context, client klient.Client, caaPod v1.Pod, Pod v1.Pod) (string, error) {
+	pullingtime := ""
+	podLogString := ""
+	var startTime, endTime time.Time
+	clientset, err := kubernetes.NewForConfig(client.RESTConfig())
+	if err != nil {
+		return "", err
+	}
+
+	if Pod.Status.Phase == v1.PodRunning {
+		req := clientset.CoreV1().Pods(caaPod.ObjectMeta.Namespace).GetLogs(caaPod.ObjectMeta.Name, &v1.PodLogOptions{})
+		podLogs, err := req.Stream(ctx)
+		if err != nil {
+			return "", err
+		}
+		defer podLogs.Close()
+		buf := new(bytes.Buffer)
+		_, err = io.Copy(buf, podLogs)
+		if err != nil {
+			return "", err
+		}
+		podLogString = buf.String()
+
+		if podLogString != "" {
+			podLogSlice := reverseSlice(strings.Split(podLogString, "\n"))
+			for _, i := range podLogSlice {
+				if strings.Contains(i, "calling PullImage for \""+Pod.Spec.Containers[0].Image+"\"") {
+					startTime, err = time.Parse("15:04:05", strings.Split(i, " ")[1])
```

Comment:
I wonder if using regex, or `scan_fmt` for this matching and extraction the time string might cleaner?